### ISSUE TITLE
Change features_only to feature_defs_only

### ISSUE THREAD

**Issue by dvreed77** [alteryx/featuretools#1867]
When looking at the [DFS](https://featuretools.alteryx.com/en/stable/generated/featuretools.dfs.html#featuretools-dfs) call, I feel that the `features_only` option is misleading. Setting this to `True` only returns definitions and not the feature matrix. So I believe the option should be `feature_defs_only`

#### Code Example

```python
import featuretools as ft

es = ft.demo.load_mock_customer(return_entityset=True)

feature_defs = ft.dfs(
    entityset=es,
    target_dataframe_name="customers",
    agg_primitives=["mean"],
    trans_primitives=["time_since_previous"],
    features_defs_only=True,
)

feature_defs

```

**Comment by dvreed77** [comment]
Need to update docs for this as well. 

definitions is misspelled here:

> The list of generated feature defintions, and the feature matrix. If features_only is True, the feature matrix will not be generated.

**Comment by Eunoia1729** [comment]
Greetings @dvreed77 :wave:, can I work on this issue ?
I hope this will help me get started in contributing to featuretools.

**Comment by thehomebrewnerd** [comment]
@Eunoia1729 Let us discuss this issue a little more before you start working on it. This would be a breaking change to the API, which would have implications in a lot of different places.

**Comment by Eunoia1729** [comment]
@thehomebrewnerd Sure ! I'll connect with you over slack

**Comment by thehomebrewnerd** [comment]
@Eunoia1729 We have decided to wait on implementing this change for now and will perhaps include this update in the future if we need to make other breaking changes. Please do not work on this issue for now.

If you want to work on another issue instead, this might be a good one to try: https://github.com/alteryx/featuretools/issues/1613

**Comment by Eunoia1729** [comment]
@thehomebrewnerd Thank you for the update ! As per your suggestion, I'll take a look into #1613 to get started.
